content/articles/2017-08-ggplot2-internship.Rmd

```diff
@@ -0,0 +1,87 @@
+---
+title: The summer of ggplot2
+slug: ggplot2-internship
+description: >
+  This summer I had the great fortune to accept the ggplot2 internship baton and
+  spend ten weeks developing new features and fixing bugs in ggplot2.
+author: Kara Woo
+date: 2017-08-25
+photo: 
+  url: https://unsplash.com/photos/DpF8nI-dQKA
+  author: Joe Shillington
+categories: [other]
+---
+
+This summer I had the great fortune to accept the
+[ggplot2 internship baton](https://twitter.com/dataandme/status/880952970602532864)
+from [Thomas Lin Pedersen](http://www.data-imaginist.com/2016/Becoming-the-intern/)
+and spend ten weeks developing new features and fixing bugs in ggplot2. My
+internship was a great experience, and I learned a ton from working with Hadley
+and digging into the ggplot2 codebase.
+
+## Daily life as a ggplot2 intern
+
+My ggplot2 work consisted of a few different things: issue and pull request
+triage, bug fixes, and developing new features. The tidyverse gets a lot of new
+issues opened on GitHub, so to help keep things organized I tagged each ggplot2
+issue with labels describing the type of issue and which part of ggplot2 it
+related to (scales, themes, layers, etc.). In some cases I helped the authors
+create reprexes so that we could diagnose the problems and determine when they
+were fixed. I triaged new pull requests as well, merging anything that was very
+minor, helping authors conform to the tidyverse style guide, and requesting
+reviews from Hadley for bigger changes.
+
+Fixing bugs was fun detective work and helped me build an understanding of how
+ggplot2 works internally. Along the way I got pretty good at using R's debugging
+tools and have come to really appreciate the value of following a consistent
+[coding style](http://style.tidyverse.org/). It is vastly easier to understand
+what a piece of code is doing when it is written in a readable and consistent
+way. Enforcing a style guide is time very well spent.
+
+I implemented some significant new features during this summer as well. The
+first was an overhaul of certain scale types in ggplot2. Scales are what control
+how data gets mapped to visual elements on a plot, and they vary based on the
+type of data (for instance, you can't map a continuous variable to discrete
+shapes). As part of this project I improved support for datetime scales and
+ordered factors. I also added built-in support for [viridis color
+palettes](https://bids.github.io/colormap/), a move that has garnered me some
+significant Twitter popularity. `r emo::ji("sunglasses")`
+
+I created a new position function, `position_dodge2()`, for placing box plots,
+bars, and rectangles. This project started out as a fix for a bug where boxes of
+differing widths could not be dodged from one another using the existing
+`position_dodge()` function, but it quickly grew into a larger project that
+fixed not only the original issue, but three other issues as well.
```

Comment:
Do you want to link to any that? I can understand if the answer is "no". But if you're neutral to positive, I think it helps other people get a very concrete sense of what a bug is, what a bug fix looks like, and how different bugs end up have common causes/fixes.